### PR TITLE
Temperature streaming from the BatteryPool

### LINKED ISSUE #466: `BatteryPool` should expose a temperature receiver
### What's needed?

By migrating away from `PowerDistributor` and use of `BatteryData` / `InverterData`, the only thing missing is access to the `temperature` of a battery(-pool).

### Proposed solution

`BatteryPool.temperature` should stream objects that contain the min/max/avg block temperatures of all batteries in the pool.  The returned objects could implement `__iter__`, so they can be directly used in a loop.  But it is nice to use an class, rather than a list, because later on we can add more methods for finding the hottest block, avg of all batteries, etc.

The fields could probably be called {temp_lowest/temp_highest/temp_avg} to indicate they are not bounds, but actual measurements.

This can be implemented with the same approach as `soc` and `capacity`.

### Use cases

Analogous to `soc`, `power`, `power_bounds`, `capacity`, add a `temperature` receiver.

### Alternatives and workarounds

still depend on `BatteryData` and legacy APIs :)

### Additional context

This is my second attempt to migrate our PQ actor to the BatteryPool :)

### PR DESCRIPTION
**Pull request by shsms**
Closes #466

### CODE REVIEW

**Review by llucax — COMMENTED** [review]
LGTM, needs release notes.

**Comment by shsms** [comment]
> LGTM, needs release notes.

done

File: src/frequenz/sdk/microgrid/component/_component_data.py [hunk 1]
```diff
@@ -141,8 +141,8 @@ class BatteryData(ComponentData):
         This will be a positive number, or zero if no charging is possible.
     """
 
-    temperature_max: float
-    """The maximum temperature of all the blocks in a battery, in Celcius (°C)."""
+    temperature: float
+    """The (average) temperature reported by the battery a battery, in Celcius (°C)."""
```
Comment (llucax) [inline src/frequenz/sdk/microgrid/component/_component_data.py]:
```suggestion
    """The (average) temperature reported by the battery, in Celcius (°C)."""
```

**Review by llucax — COMMENTED** [review]
Sorry, just discovered some typo/left-over.

As a side note I'm also still wondering if we don't want to explicitly say that this is average temperature, but I guess this was already discussed and was agreed to leave it without?

**Comment by shsms** [comment]
> As a side note I'm also still wondering if we don't want to explicitly say that this is average temperature, but I guess this was already discussed and was agreed to leave it without?

do you mean in the field name?  because it says average in the docs: ```The (average) temperature reported by the battery, in Celcius (°C)```

**Comment by llucax** [comment]
Yeah, field name.

**Review by llucax — DISMISSED** [review]
Approving as I think it is very likely that we don't want to change it, and if we do it is easier to do in a followup PR.

**Comment by shsms** [comment]
Had to merge 0.x.x because of conflict.  Couldn't rebase, because the PR already has a merge commit.

**Review by llucax — APPROVED** [review]

**Pull request merged**